lib/spack/llnl/util/filesystem.py

```diff
@@ -1819,3 +1820,20 @@ def remove_directory_contents(dir):
                 os.unlink(entry)
             else:
                 shutil.rmtree(entry)
+
+
+@contextmanager
+def keep_modification_time(*filenames):
```

Comment:
I'd prefer that users just pass in the exact file paths they want to preserve (and the user can use a glob to construct that set of files if they want).